rollup.config.mjs

```diff
@@ -56,45 +51,88 @@ const node_build = {
  *
- * The bundle provides a globally available `strapi` variable and includes source maps for debugging.
+ * Outputs:
+ * - CommonJS (dist/bundle.browser.cjs): for environments using `require`.
+ * - ES Module (dist/bundle.browser.mjs): for environments with native support for ES Modules.
  *
- * The bundle is minified for better performance.
+ * @type {import('rollup').RollupOptions}
+ */
+const browser_build = {
+  input: 'src/exports.ts',
+  cache: true,
+  output: [
+    // CommonJS Browser build
+    {
+      file: 'dist/bundle.browser.cjs',
+      format: 'cjs',
+      sourcemap: isProduction ? 'hidden' : true,
+      exports: 'named',
+    },
+    // ESM Browser build
+    {
+      file: 'dist/bundle.browser.mjs',
+      format: 'esm',
+      sourcemap: isProduction ? 'hidden' : true,
+      exports: 'named',
+    },
+  ],
+  external: ['debug'],
+  plugins: [
+    // Locate modules using the Node resolution algorithm, for using third party modules in node_modules
+    nodeResolve({
+      browser: true, // Only resolve browser-compatible modules
+      preferBuiltins: false, // Disable Node.js built-ins
+    }),
+    // Convert CommonJS modules to ES6, so they can be included in a Rollup bundle
+    commonjs(),
+    // Transpile TypeScript to JavaScript
+    typescript({ tsconfig: './tsconfig.build.json' }),
+    // Replace environment variables and add process.browser
+    replace({
+      'process.browser': true,
+      preventAssignment: true,
+    }),
+    // Minify the output in production
+    isProduction && terser(),
+  ],
+};
+
+/**
+ * This configuration is designed to create IIFE format bundles suitable for direct
+ * inclusion in web browsers.
  *
  * Outputs:
- * - IIFE Minified (dist/bundle.browser.min.js): a minified browser build, optimized for performance, with source maps.
+ * - IIFE Minified (dist/bundle.browser.min.js): Minified JavaScript file containing the entire bundle, optimized
+ *   for direct browser usage and embedding, exposed with the global `strapi` variable.
  *
- * Plugins Used:
- * - `nodeResolve`: resolves browser-compatible modules from `node_modules` while ignoring Node.js built-ins.
- * - `commonjs`: converts CommonJS modules (for example, dependencies) to ES modules for bundling.
- * - `typescript`: transpiles TypeScript code into JavaScript to ensure compatibility with browsers.
- * - `replace`: replaces `process.env` variables and includes a `process.browser` variable for browser-specific code.
- * - `terser`: minifies the production output to improve loading performance and reduce file size.
+ * External Dependencies:
+ * - `debug` is excluded from the bundle and replaced with a no-op function.
  *
  * @type {import('rollup').RollupOptions}
  */
-const browser_build = {
+const iife_build = {
   input: 'src/exports.ts',
   cache: true,
   output: [
     {
-      file: 'dist/bundle.browser.min.js',
+      file: 'dist/bundle.iife.min.js',
```

Comment:
is it worth keeping browser in there like `dist/bundle.browser.iife.min.js` ? It's a little confusing having the package.json browser attribute pointing to something without browser in it